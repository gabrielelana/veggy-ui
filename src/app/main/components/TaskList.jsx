import moment from 'moment'

const TaskList = ({timers}) => {
  const rows = timers.map(t => <TaskRow key={t.id} {...t} />)
  return (
    <div className="column is-one-third">
      <table className="table">
        <thead>
          <tr>
            <th>Started at</th>
            <th>Description</th>
            <th>Shared</th>
            <th className="centered">Status</th>
          </tr>
        </thead>
        <tbody>
         {rows}
        </tbody>
      </table>
    </div>
  )
}

TaskList.defaultProps = {
  timers: []
}

function TaskRow(props){
  var shared_with = props.shared_with.map(w => <span className="tag is-info" key={w} >{`${w}`}</span>)
  return (
    <tr>
      <td>{moment(props.startedAt).format('DD-MM-YYYY hh:mm:ss')}</td>
      <td>{props.description}</td>
      <td>{shared_with}</td>
      <td className="centered">{getStatusIcon(props.status)}</td>
    </tr>)
}

const statuses = {
  'started': 'pi-pomodoro-ticking',
  'completed': 'pi-pomodoro-done',
  'squashed': 'pi-pomodoro-squashed'
}

function getStatusIcon(status){
  return <span className="icon"><i className={statuses[status]}></i></span>  
}

export default TaskList