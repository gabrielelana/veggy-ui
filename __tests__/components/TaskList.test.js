import React from 'react'
import renderer from 'react-test-renderer'
import TaskList from '../../src/app/main/components/TaskList'

describe('<TaskList />', () => {
  
  test('should display the tasks', () => {
    const timers = [{
      id: 1,
      startedAt: '2010-04-09 11:10',
      shared_with: [],
      status: 'completed'
    },{
      id: 2,
      startedAt: '2010-04-11 22:20',
      shared_with: [],
    }]
    const component = renderer.create(<TaskList timers={timers} />)
    expect(component.toJSON()).toMatchSnapshot()
    
  })
})