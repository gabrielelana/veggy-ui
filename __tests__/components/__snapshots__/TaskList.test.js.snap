// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TaskList /> should display the tasks 1`] = `
<div
  className="column is-one-third"
>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          Started at
        </th>
        <th>
          Shared
        </th>
        <th
          className="centered"
        >
          Status
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          09-04-2010 11:10:00
        </td>
        <td />
        <td
          className="centered"
        >
          <span
            className="icon"
          >
            <i
              className="pi-pomodoro-done"
            />
          </span>
        </td>
      </tr>
      <tr>
        <td>
          11-04-2010 10:20:00
        </td>
        <td />
        <td
          className="centered"
        >
          <span
            className="icon"
          >
            <i
              className={undefined}
            />
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
